doc improvements (#12)
LLNL/radiuss-shared-ci: docs/sphinx/user_guide/setup_ci.rst

The third step in adopting RADIUSS Shared CI infrastructure is to setup the CI.

Once you have put in the effort to implement the first two steps, you should be
able to benefit from the shared CI infrastructure. In more complex scenarios,
you will always be able to use the template as a starting point for a custom
implementation.

By sharing the CI definition, projects share the burden of maintaining it. In
addition, with our shared CI, they also share a core set of Spack specs to
ensure that they keep running tests with similar configurations.

=================
RADIUSS Shared CI
=================

Initially, the RADIUSS effort developed a mechanism to 
`share Spack configuration files`_ across projects. Then, we developed a 
process to manage Spack and use it to generate CMake configuration files.
The RADIUSS Shared CI project extends this by developing a CI implementation
that can be shared across projects, which drive it with a ``build-and-test`` 
script that has the same inputs across projects.

> With a centralized CI configuration that is shared by projects, we try to keep 
> its interface minimal, while allowing for project-specific customization.

Author: ```suggestion
With a centralized CI configuration that is shared by projects, we create an interface between local and shared configuration. we try to keep this interface minimal, while allowing for project-specific customization. Only a handful of modifications are required to get the CI to work for your projects, while files in the ``customization`` directory allow for finer tuning and extensibility.
```

By default, GitLab expects a ``.gitlab-ci.yml`` file to interpret the CI setup.
We provide one in ``customization/gitlab-ci.yml`` that projects can copy-paste.
Make sure to place the file in the top-level directory of your project, and
that it has a dot (``.``) at the beginning of the name.

.. code-block:: bash

   cp ../radiuss-shared-ci/customization/gitlab-ci.yml .gitlab-ci.yml

In the ``.gitlab-ci.yml`` file, there are some variables you need to adapt to 
your project. They are described in the following table:

 ========================================== ==========================================================================================================================
  Parameter                                  Description
 ========================================== ==========================================================================================================================

>   ``LLNL_SERVICE_USER``                      Service Account for project used in CI

Author: ```suggestion
  ``LLNL_SERVICE_USER``                      Project specific Service Account used in CI
```

   If a variable is blank in the template file, then it does not require a
   value. If a variable has a value there, it does require one.

You may add configurations to the ``.custom_build_and_test`` job that will then

> be included in all you CI jobs. This can be used `export jUnit test reports`_,

Author: ```suggestion
be included in all you CI jobs. This can be used to `export jUnit test reports`_,
```
